packages/consumption/src/modules/attributes/AttributesController.ts

```diff
@@ -964,9 +975,32 @@ export class AttributesController extends ConsumptionBaseController {
             query["shareInfo.sourceAttribute"] = repositoryAttribute.id.toString();
             const sharedCopies = await this.getLocalAttributes(query);
 
-            ownSharedIdentityAttributeVersions.push(...sharedCopies);
+            ownSharedIdentityAttributePredecessors.push(...sharedCopies);
         }
 
-        return ownSharedIdentityAttributeVersions;
+        return ownSharedIdentityAttributePredecessors;
+    }
+
+    public async getSharedSuccessorsOfRepositoryAttribute(repositoryAttribute: LocalAttribute, query?: any): Promise<LocalAttribute[]> {
+        if (typeof query === "undefined") {
+            query = {};
+        }
+
+        const ownSharedIdentityAttributeSuccessors: LocalAttribute[] = [];
+        while (repositoryAttribute.succeededBy) {
```

Comment:
Use the `typeof foo !== "undefined"` syntax. Using a different construct here suggests that you're doing something else than just checking for `undefined`. Same for the implementation for `getSharedPredecessorsOfRepositoryAttribute`.